activerecord/lib/active_record/type/date.rb

```diff
@@ -9,6 +9,10 @@ def klass
         ::Date
       end
 
+      def type_cast_for_database(value)
```

Comment:
Yes, after further investigation I determined we need to just type cast the values ourselves. Thanks @kamipo